packages/core/strapi/package.json

```diff
@@ -129,14 +131,17 @@
     "open": "8.4.0",
     "ora": "5.4.1",
     "package-json": "7.0.0",
+    "pkg-up": "3.1.0",
     "qs": "6.11.1",
     "resolve-cwd": "3.0.0",
     "semver": "7.5.4",
-    "statuses": "2.0.1"
+    "statuses": "2.0.1",
+    "typescript": "5.1.3",
+    "vite": "4.4.9",
```

Comment:
I am wondering if `vite` should be a peer dependency and listed as a top-level dependency in a plugin? This would allow our users to upgrade vite without having to wait for us to do it.

This is probably a bigger discussion ...

Reply:
I wouldn't do that, because we use a complete config if they were to prematurely update to a breaking change then it could fail. IMO it's fine to be listed as a dependency, people don't have to use this CLI if they don't want too 😄

Reply:
I think this is more of a general question also for using `vite` as bundler in the end. I feel like most vite projects have `vite` in the top-level, because it allows developers to write a custom vite configuration, which then often is just merged with the one the applications brings in by default.

We could still specify the supported version range (minor + patch versions).

Reply:
I don't think we're gonna allow people to just write a custom `vite.config.js` file because realistically if we're making a bundler tool like this we want to support _more_ stuff e.g. it's tricky to do a CLI tool this way. I think instead we'd support a custom package config e.g `pkg.config.js` and include something like `viteFinal` similar to storybook

Reply:
We could, but we could probably also make Strapi a plugin:

```
// vite.config.js

import { strapiPlugin } from '@strapi/strapi/vite';

export default {
  plugins: [strapiPlugin()] // strapiPlugin would also load `vite-react`
}
```

I feel like this is more common in vite land, but we can discuss this again once we get to bundling.

Reply:
We could, but we could probably also make Strapi a plugin:

```
// vite.config.js

import { strapiPlugin } from '@strapi/strapi/vite';

export default {
  plugins: [strapiPlugin()] // strapiPlugin would also load `vite-react`
}
```

I feel like this is more common in vite land, but we can discuss this again once we get to bundling.